atomic_reactor/inner.py

```diff
@@ -324,11 +324,11 @@ def load(cls, data: Dict[str, Any]):
             tag_conf.add_floating_image(image)
         return tag_conf
 
-    def dump(self) -> Dict[str, Any]:
+    def as_dict(self) -> Dict[str, Any]:
         return {
-            "primary_images": [image.to_str() for image in self._primary_images],
-            "unique_images": [image.to_str() for image in self._unique_images],
-            "floating_images": [image.to_str() for image in self.floating_images],
+            "primary_images": self.primary_images,
+            "unique_images": self.unique_images,
+            "floating_images": self.floating_images,
```

Comment:
Should the `load()` method be changed accordingly? (or at least the type annotations for it: images are now `ImageName`, not `str`)

Reply:
Good catch. The magic is `ImageName.parse` actually accepts either `str` or `ImageName` object. Changing the type annotation should be enough I think.